### PR TITLE
Refactoring/remove make old model definition

### PR DESCRIPTION
**Pull request by x3medima17**
### Description of the Change
Removes primitive and any definition of old model 
Depends on #1348 

### Benefits
No unused methods

### Possible Drawbacks 
None

### CODE REVIEW

File: shared_model/cryptography/keypair.hpp [hunk 1]
```diff
@@ -32,11 +31,8 @@ namespace shared_model {
     /**
      * Class for holding a keypair: public key and private key
      */
-#ifndef DISABLE_BACKWARD
-    class Keypair : public interface::Primitive<Keypair, KeypairOldModelType> {
-#else
+
```
Comment (muratovv) [inline shared_model/cryptography/keypair.hpp]: Redundant endline

File: shared_model/interfaces/base/signable.hpp [hunk 2]
```diff
@@ -36,27 +36,14 @@ namespace shared_model {
...
   namespace interface {
 
-#ifdef DISABLE_BACKWARD
-#define SIGNABLE(Model) Signable<Model>
-#else
-#define SIGNABLE(Model) Signable<Model, iroha::model::Model>
-#endif
+    /**
+     * Interface provides signatures and adds them to model object
+     * @tparam Model - your new style model
```
Comment (muratovv) [inline shared_model/interfaces/base/signable.hpp]: Already not a "new" style.

File: shared_model/interfaces/queries/get_account_assets.hpp [hunk 3]
```diff
@@ -18,19 +18,14 @@
 #ifndef IROHA_SHARED_MODEL_GET_ACCOUNT_ASSETS_HPP
 #define IROHA_SHARED_MODEL_GET_ACCOUNT_ASSETS_HPP
 
-#include "interfaces/base/primitive.hpp"
```
Comment (muratovv) [inline shared_model/interfaces/queries/get_account_assets.hpp]: Seems like primitive removed but model primitive doesn't use here.

**Comment by muratovv** [comment]
Also, please, remove DISABLE_COMPATIBILITY from shared_model ::cmake and ::bindings.

File: shared_model/interfaces/queries/get_transactions.hpp [hunk 4]
```diff
@@ -18,19 +18,14 @@
 #ifndef IROHA_SHARED_MODEL_GET_TRANSACTIONS_HPP
 #define IROHA_SHARED_MODEL_GET_TRANSACTIONS_HPP
 
-#ifndef DISABLE_BACKWARD
-#include "model/queries/get_transactions.hpp"
-#endif
-
 #include <boost/range/algorithm/for_each.hpp>
```
Comment (lebdron) [inline shared_model/interfaces/queries/get_transactions.hpp]: Please remove, it is unused.

File: shared_model/interfaces/query_responses/error_responses/no_signatories_error_response.hpp [hunk 5]
```diff
@@ -33,13 +33,6 @@ namespace shared_model {
       std::string reason() const override {
         return "NoSignatoriesErrorResponse";
       }
-
```
Comment (lebdron) [inline shared_model/interfaces/query_responses/error_responses/no_signatories_error_response.hpp]: Documentation is wrong on line 28 🙈

File: shared_model/interfaces/query_responses/error_responses/not_supported_error_response.hpp [hunk 6]
```diff
@@ -33,13 +33,6 @@ namespace shared_model {
       std::string reason() const override {
         return "NotSupportedErrorResponse";
       }
-
```
Comment (lebdron) [inline shared_model/interfaces/query_responses/error_responses/not_supported_error_response.hpp]: Same with line 28 🙈

File: shared_model/interfaces/query_responses/account_response.hpp [hunk 7]
```diff
@@ -19,22 +19,18 @@
 #define IROHA_SHARED_MODEL_ACCOUNT_RESPONSE_HPP
 
 #include <new>
```
Comment (lebdron) [inline shared_model/interfaces/query_responses/account_response.hpp]: Please remove

File: shared_model/interfaces/query_responses/asset_response.hpp [hunk 8]
```diff
@@ -19,22 +19,18 @@
 #define IROHA_SHARED_MODEL_ASSET_RESPONSE_HPP
 
 #include <new>
```
Comment (lebdron) [inline shared_model/interfaces/query_responses/asset_response.hpp]: Please remove

File: shared_model/interfaces/query_responses/signatories_response.hpp [hunk 9]
```diff
@@ -18,21 +18,16 @@
 #ifndef IROHA_SHARED_MODEL_SIGNATORIES_RESPONSE_HPP
 #define IROHA_SHARED_MODEL_SIGNATORIES_RESPONSE_HPP
 
-#include "interfaces/base/primitive.hpp"
 #include "interfaces/common_objects/types.hpp"
 #include "utils/string_builder.hpp"
 #include "utils/visitor_apply_for_all.hpp"
```
Comment (lebdron) [inline shared_model/interfaces/query_responses/signatories_response.hpp]: Visitor is not used here, please remove.

File: shared_model/interfaces/query_responses/transactions_response.hpp [hunk 10]
```diff
@@ -18,23 +18,18 @@
 #ifndef IROHA_SHARED_MODEL_TRANSACTIONS_RESPONSE_HPP
 #define IROHA_SHARED_MODEL_TRANSACTIONS_RESPONSE_HPP
 
-#include "interfaces/base/primitive.hpp"
 #include "interfaces/common_objects/types.hpp"
 #include "interfaces/transaction.hpp"
 #include "utils/polymorphic_wrapper.hpp"
 #include "utils/string_builder.hpp"
 #include "utils/visitor_apply_for_all.hpp"
```
Comment (lebdron) [inline shared_model/interfaces/query_responses/transactions_response.hpp]: Visito, transaction, polymorphic wrapper are not used, please remove.

File: shared_model/interfaces/transaction_responses/stateful_failed_tx_response.hpp [hunk 11]
```diff
@@ -18,13 +18,8 @@
 #ifndef IROHA_STATEFUL_FAILED_TX_RESPONSE_HPP
 #define IROHA_STATEFUL_FAILED_TX_RESPONSE_HPP
 
-#include "interfaces/base/primitive.hpp"
 #include "utils/string_builder.hpp"
```
Comment (lebdron) [inline shared_model/interfaces/transaction_responses/stateful_failed_tx_response.hpp]: Include not used, please remove.

File: shared_model/interfaces/transaction_responses/stateful_valid_tx_response.hpp [hunk 12]
```diff
@@ -18,13 +18,8 @@
 #ifndef IROHA_STATEFUL_VALID_TX_RESPONSE_HPP
 #define IROHA_STATEFUL_VALID_TX_RESPONSE_HPP
 
-#include "interfaces/base/primitive.hpp"
 #include "utils/string_builder.hpp"
```
Comment (lebdron) [inline shared_model/interfaces/transaction_responses/stateful_valid_tx_response.hpp]: Same for unused include

**Pull request merged**